internal/services/containerapps/helpers/container_app_environment.go

```diff
@@ -69,42 +87,49 @@ func ExpandWorkloadProfiles(input []WorkloadProfileModel) *[]managedenvironments
 
 	result := make([]managedenvironments.WorkloadProfile, 0)
 
+	consumptionDefined := false
+
 	for _, v := range input {
 		r := managedenvironments.WorkloadProfile{
-			Name:                v.Name,
-			WorkloadProfileType: v.WorkloadProfileType,
+			Name: v.Name,
 		}
 
-		if v.Name != consumption {
-			r.MaximumCount = pointer.To(int64(v.MaximumCount))
-			r.MinimumCount = pointer.To(int64(v.MinimumCount))
+		if !strings.EqualFold(v.Name, string(WorkloadProfileSkuConsumption)) {
```

Comment:
(same here) this is coming from a user specified value in the Schema

```suggestion
		if v.Name != string(WorkloadProfileSkuConsumption) {
```